### PR TITLE
XmlConfiguration refactoring

### PR DESCRIPTION
**Pull request by rollenes**
I found XmlConfiguration poorly legible, and not tested, so I made some refactorings.
Step by step I moved all static properities away from XmlConfiguration. 
A lot of testing has ben done in this area.
This is the first part of refactoring XmlConfiguration. In second part I want to move away all static methods from XmlConfiguration by introducing some kind of ConfigurationBuilder.

### CODE REVIEW

**Comment by padraic** [comment]
That's a big chunk of work @rollenes. Very much appreciated! Let me know when you're ready to merge.

**Comment by GrahamCampbell** [comment]
Probably needs squashing at the end though. :)

File: src/Adapter/Phpunit/ConfigurationLoader.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<?php
+
```
Comment (samsonasik) [inline src/Adapter/Phpunit/ConfigurationLoader.php]: missing license header

**Comment by rollenes** [comment]
Rebased and ready to merge.

**Pull request merged**